runners/google-cloud-dataflow-java/worker/src/main/java/org/apache/beam/runners/dataflow/worker/streaming/Work.java

```diff
@@ -97,6 +98,10 @@ public void setState(State state) {
     this.currentState = TimedState.create(state, now);
   }
 
+  public void setFailed(boolean isFailed) {
```

Comment:
how about removing bool param, and having this just transition to this.failed= True?
We could drop data possibly if we set failed, stopped processing, and then cleared it and committed.

Reply:
Done.